TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/QueryableExtensions.cs

```diff
@@ -0,0 +1,71 @@
+using System.Collections;
+using System.Linq.Expressions;
+
+namespace TeachingRecordSystem.SupportUi;
+
+public static class QueryableExtensions
+{
+    public static IOrderedQueryable<TSource> OrderBy<TSource, TKey>(
+        this IQueryable<TSource> source,
+        SortDirection direction,
+        Expression<Func<TSource, TKey>> keySelector)
+    {
+        return direction == SortDirection.Ascending ? source.OrderBy(keySelector) : source.OrderByDescending(keySelector);
+    }
+
+    public static IOrderedQueryable<TSource> ThenBy<TSource, TKey>(
+        this IOrderedQueryable<TSource> source,
+        SortDirection direction,
+        Expression<Func<TSource, TKey>> keySelector)
+    {
+        return direction == SortDirection.Ascending ? source.ThenBy(keySelector) : source.ThenByDescending(keySelector);
+    }
+
+    public static async Task<ResultPage<T>> GetPageAsync<T>(
+        this IQueryable<T> source,
+        int? currentPage,
+        int itemsPerPage)
+    {
+        var page = Math.Max(currentPage ?? 1, 1);
+
+        var total = await source.CountAsync();
```

Comment:
There was a reason this count was part of the `Filters` class, it calculates the total item count as part of calculating the counts for all the filter options

Reply:
Also if this is used with filters, then calling `CountAsync` here will create a third query - one to calculate the filter counts, one to get the current page of items, and this one to calculate the total count

Reply:
Yeh - I'm starting to think that the filters implementation and the simpler one here can't use the same implementation...

Reply:
I wonder if something like a querybuilder would work where it knows whether or it has filters, and either delegates to the filter count or does a `CountAsync`